### PR TITLE
Ability to read/write indicator level (of a specific plan) via REST API

### PR DESCRIPTION
**Pull request by tituomin**
## Description
Ability to read/write indicator levels (of a specific plan) via REST API

## Screenshots/Videos (if applicable)
N/A

## Requirements, dependencies and related PRs
https://app.asana.com/1/1201243246741462/project/1206017643443542/task/1206286256572076?focus=true

## Additional Notes
This is based on the fact that an indicator is exposed nested under a plan. Hence, the field "level" is in singular and it only exposes the level of that specific plan, if any.

------

### Type of Change
- [X] Set the PR's label to match the nature of this change

### Testing
- [X] **Built Unit tests** (unit tests added/updated)
- [ ] **Built E2E tests** (if applicable. E2E tests added/updated)
- [X] **Authorization is tested** (permissions and access controls verified)
- [X] **Manually tested locally** (functionality verified)
    ##### Manual testing instructions
    If feature requires manual testing by reviewer, you can provide instructions here.

### Internationalization & Accessibility
- [X] **New strings are translatable** (all user-facing text uses i18n)
- [X] **Accessibility** standards met (WCAG compliance, screen reader support)

### Integrations (if applicable)
If there are model changes to models which use any of the features below, verify the new models work together with the features.
For example, when adding a new model, verify the new model instances are copied when copying a plan.
- [X] **Moderation** integration implemented
- [X] **Reporting** integration implemented
- [X] **Copy plan feature** integration implemented

### Dependencies
- [X] **Dependencies are merged** (if applicable. If the change depends on other PRs e.g. kausal_common)

### CODE REVIEW

**Comment by tituomin** [comment]
> ## Code review
> I found 2 issues that need attention:
> 
> ### 1. Missing type annotation on `update()` method parameter (`indicators/api.py:518`)
> The `update()` method is missing a type annotation on the `validated_data` parameter. According to [CLAUDE.md](https://github.com/kausaltech/kausal-watch/blob/7e3422d7a79d2f24dac6622eb8fa67ab79e4e4af/CLAUDE.md#L105-L108), you should use modern Python 3.13+ type annotation syntax.
> 
> **Current code:**
> 
> https://github.com/kausaltech/kausal-watch/blob/7e3422d7a79d2f24dac6622eb8fa67ab79e4e4af/indicators/api.py#L517-L519
> 
> **Issue:** The `create()` method already has the proper annotation (`validated_data: dict`), but `update()` does not.
> 
> **Fix:** Add type annotation: `def update(self, instance, validated_data: dict):`
> 
> **CLAUDE.md reference:**
> 
> > Use modern Python 3.13+ type annotation syntax:
> > 
> > * `dict` instead of `Dict`, `list` instead of `List`
> 
> ### 2. Bug: Update method deletes level when field is omitted from request (`indicators/api.py:521`)
> The `update()` method has a logic bug where it deletes the indicator's level when the `level` field is simply omitted from the update request payload.
> 
> **Current code:**
> 
> https://github.com/kausaltech/kausal-watch/blob/7e3422d7a79d2f24dac6622eb8fa67ab79e4e4af/indicators/api.py#L520-L537
> 
> **The problem:**
> 
> * Line 521: `level = validated_data.pop('level', None)` defaults to `None` when the field is not provided
> * Lines 533-534: `if level is None: instance.levels.filter(plan=plan).delete()` deletes the level
> * This means any PUT/PATCH request that doesn't include the `level` field will inadvertently delete the existing level
> 
> **Inconsistency:** The `create()` method correctly handles this by using a sentinel value to distinguish "not provided" from "explicitly None":
> 
> https://github.com/kausaltech/kausal-watch/blob/7e3422d7a79d2f24dac6622eb8fa67ab79e4e4af/indicators/api.py#L495-L506
> 
> **Fix:** Use the same sentinel pattern in `update()`:
> 
> ```python
> _not_provided = object()
> level = validated_data.pop('level', _not_provided)
> # ...
> if level is _not_provided:
>     pass  # Leave level unchanged
> elif level is None:
>     instance.levels.filter(plan=plan).delete()
> else:
>     # Update/create the level
>     try:
>         plan_level = instance.levels.get(plan=plan)
>     except ObjectDoesNotExist:
>         instance.levels.create(plan=plan, level=level)
>     else:
>         plan_level.level = level
>         plan_level.save()
> ```
> 
> **Missing test coverage:** None of the current tests verify that omitting the `level` field preserves the existing level. Consider adding a test case for this scenario.
> 
> Checked for bugs and CLAUDE.md compliance.

Fixed.

File: indicators/tests/test_api.py [hunk 1]
```diff
@@ -464,3 +464,265 @@ def test_bulk_indicator_put_creates_individual_log_entries(
...
+        'level': 'tactical',
+    })
+    assert response.status_code == 200
+
+    indicator.refresh_from_db()
+    level = indicator.levels.get(plan=plan)
+    assert level.level == 'tactical'
+
+
+def test_update_indicator_level_to_null_removes_level(api_client, plan, plan_admin_user, indicator, indicator_detail_url):
```
Comment (Bhavatu) [inline indicators/tests/test_api.py:580]: Isn't this testing pretty much the same thing as [test_update_indicator_with_null_level_deletes_existing_level](https://github.com/kausaltech/kausal-watch/pull/492/changes#diff-22921254a18949e9aa76ff49f863baffb511431ed8aab4292fa389b1b68a3d75R710) or am I missing something?

**Review by Bhavatu — APPROVED** [review]
Just a minor comment.
